src/plugins/locale/index.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable dot-notation */
+import type { EsDayPlugin, Locale, UnitType } from 'esday'
+import * as C from '~/core/constant'
+import { prettyUnit, undefinedOr } from '~/core/utils'
+import en from '~/locales/en'
+
+const LocaleStore: Map<string, Locale> = new Map()
+
+let $localeGlobal = 'en'
+
+export function getLocale(name: string): Locale {
+  return LocaleStore.get(name) || en
+}
+
+export function registerLocale(locale: Locale, rename?: string): void {
```

Comment:
for me `rename` sounds more like a boolean element (rename yes or no). How about something like `newName` or simply `name`?